Catappult/appcoins-wallet-android
# [APPC-2693] Wallet Log Upload

**What does this PR do?**

Adds Wallet Log Upload feature to settings menu.

**Database changed?**

   Yes.

**Where should the reviewer start?**

- [ ] SettingsPresenter.kt
- [ ] SendLogsRepository.kt

**How should this be manually tested?**

Generate an error during an IAP for example, set that your wallet can log using the backend backoffice, and then send the log file using the option that should appear in the settings. Check if the file was received in the aws backoffice and if its contents are correct. 

  Flow on how to test this or QA Tickets related to this use-case: [APPC-2693](https://aptoide.atlassian.net/browse/APPC-2693)

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-2693](https://aptoide.atlassian.net/browse/APPC-2693)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.) Yes. `desugar_version`




**Code Review Checklist**
